### DISCUSSION TITLE
Can't get Replex working on remote access.

### DISCUSSION THREAD

**Discussion by cine25** [lostb1t/replex#74] [Q&A]
Hello, I can't get replex to run other than local:

i proceed as follows:

I start replex in docker with the following command:
ocker run --rm -it -p 3001:80 -e REPLEX_HOST="http://plexip:plexport" ghcr.io/lostb1t/replex:latest

after that I see replex working on local server. With HEROROWS and everything. Now when I disable remote access Plex works only local on server with plexip:3001 and there is no access externally, neither via app.plex.tv nor via direct plexip:3001, also clients have no connection anymore.

i have put http://plexip:3001/ in network tab. but this doesn't work either.

What am I doing wrong?

Thank you!

**Comment by lostb1t** [comment]
Localhost only works for your local pc. Extern devices won't be able to access it. Use the the lan or wan ip.

Also app.plex.tv won't work without https you need a reverse proxy for that. You can read about it in the readme.

**Reply by cine25** [reply]
I used wan ip of server where plex is running. But Clients doesn't get connection if I turn off remote access.
Its only working on the same machine where plex and replex where running.

**Reply by lostb1t** [reply]
Did you try accessing it with the wan ip through a browser to check if it's accesible? Even try it on your phone with wifi off to see if it's reachable
From outside

**Reply by cine25** [reply]
Thanks for your answer!

I Can access it with wan-IP:3001 on other PC and phone without wifi (outside). I see the Hero Rows after logout and login.
BUT clients not showing that Replex is working.
If I put wan-IP:3001 in Network Tab of plex, and disable remote access, clients not able to connect to plex anymore. then only connection what is working is browser and wan-IP:3001.

Maybe I put something wrong in Network Tab?

**Reply by lostb1t** [reply]
Hm some clients don't allow non https connections but I don't know which one. 
2 things

clear the cache of the client you are using.

Second make sure you put in the url in the custom server access urls 

If that doesn't work then I suggest using a reverse proxy with ssl

**Reply by cine25** [reply]
is this specified correctly? or should it be specified without port?

As soon as I turn off remote access no client works anymore (no connection to plex at all). I have tried it with Apple TV, Mac App and Fire TV.
<img width="473" alt="Bildschirmfoto 2023-09-07 um 18 28 34" src="https://github.com/lostb1t/replex/assets/144341972/a26c8da7-f3bb-45e3-89af-a5e21df90f10">

**Reply by lostb1t** [reply]
Yes that’s correct. Did you clear the cache? If it isn’t working then ran out of ideas. Only thing I can suggest is using a reverse proxy

**Comment by cine25** [comment]
I've been trying to get replex to run somehow for 3 weeks now ;)

Can you maybe guide me through the procedure of a reverse proxy? I have Nginx and a custom domain with SSL.

**Reply by lostb1t** [reply]
This url will tell you wat urls your server accepts. 

https://clients.plex.tv/api/v2/resources?includeIPv6=1&includeRelay=1&X-Plex-Language=en-NL&X-Plex-Token=YOURTOKEN&X-Plex-Client-Identifier=1234

Replace yourtoken with your plex token.

You can post the output here if you want me to look at it aswell (remove accestokens from output)

**Comment by cine25** [comment]


**Reply by lostb1t** [reply]
Doesn't seem to include your 3001 custom url?

**Reply by lostb1t** [reply]
Also you should remove the image. As it contained a. Accesstoken

**Reply by cine25** [reply]
hm...but I can access it over IP:3001 and it show the rows.

**Reply by lostb1t** [reply]
K I tested it myself with an ip instead of domain. And it seems plex doesn't pass it through. It should work but it doesn't. 🤷‍♂️

Your only option is setting up a proxy with a domain

I'll edit the readme

**Comment by lostb1t** [comment]
As for a reverse proxy I think the easiest ones are 

https://caddyserver.com/docs/quick-starts/reverse-proxy

Or 

https://nginxproxymanager.com

**Reply by cine25** [reply]
Is it possible to run NGINX on another machine than Replex and Plex Server?

I have tried it with this format...but it doesn't worked in my last test:

server {
    listen 80;
    server_name my-domain.com;
  location / {
        proxy_pass http://PLEXIP:32400;
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
    }
}


The Domain is A-Record pointed to Plex Server IP and SSL is activated.

**Reply by lostb1t** [reply]
Yes it can be hosted somewhere else but domain needs to be pointed to nginx. And nginx upstream should be Replex not plex

**Reply by lostb1t** [reply]
https://github.com/lostb1t/replex/discussions/71#discussioncomment-6947261

**Comment by cine25** [comment]
Okay...i have now progressed a few steps ;)

I got my Domain working with SSL and Replex is showing if I go to Domain in Web Browser.
On some Clients it is also working (fire tv stick as example). but app.plex.tv and many other clients not working? even if I logout/login and restart apps.

**Reply by lostb1t** [reply]
Use this url again https://clients.plex.tv/api/v2/resources?includeIPv6=1&includeRelay=1&X-Plex-Language=en-NL&X-Plex-Token=YOURTOKEN&X-Plex-Client-Identifier=1234

To see if plex is properly configured with your domain

**Reply by cine25** [reply]
It shows this. And app.plex.tv and most clients show server is offline.
<img width="291" alt="Bildschirmfoto 2023-09-09 um 01 24 54" src="https://github.com/lostb1t/replex/assets/144341972/4f929cd8-2cd4-4957-8df5-fbc3044f2b7e">
<img width="1506" alt="Bildschirmfoto 2023-09-09 um 01 23 00" src="https://github.com/lostb1t/replex/assets/144341972/ebbff1fc-a9f1-454a-b3c9-20a23105343f">

**Reply by cine25** [reply]
And website inspector on safari (Mac) shows this about app.plex.tv:
<img width="1509" alt="Bildschirmfoto 2023-09-09 um 01 31 07" src="https://github.com/lostb1t/replex/assets/144341972/2afd53e7-d63b-4e4a-ab71-65e3e11ce446">

**Reply by lostb1t** [reply]
Remove the port from your url in the custom server field

It seems plex is accesible from https://hq-server.online

Not https://hq-server.online:3001

**Reply by cine25** [reply]
Its the same behavior if I put port or not in custom url field.

**Reply by lostb1t** [reply]
Leave the port out. It should now work on your clients other then the web. Can you confirm? clear the cache on the client if it's not showing up


It looks like you a CORS problem in your nginx for the web app

**Reply by cine25** [reply]
I had leaved the port out, but it's not working...only on fire tv...but not app.plex.tv and all other clients. with not working I mean the complete connection to server is gone, not only replex is not working.

Yes..I have googled about this "Access-Control-Allow-Origin" problem. can it be that my server is already linked to its own domain. in my case it is : ds34965.seedhost.eu:32400

and that's why the other domain is not working properly? can i somehow work around that plex uses both domains?

**Reply by lostb1t** [reply]
Can you post your nginx config?

And honestly it should work beside app.plex.tv. I still think it's caching or sole dns issue on your other clients

What client does not work?

**Reply by cine25** [reply]
this is with nginx proxy manager:
<img width="1187" alt="Bildschirmfoto 2023-09-09 um 09 29 05" src="https://github.com/lostb1t/replex/assets/144341972/9793f3f5-adc2-4225-b4a1-1d2234fa6c18">

**Reply by cine25** [reply]
And this is the nginx standard config file:

<img width="980" alt="Bildschirmfoto 2023-09-09 um 09 35 20" src="https://github.com/lostb1t/replex/assets/144341972/5d94799c-c67d-4c02-886e-c2e583d70ff8">

**Reply by cine25** [reply]
As example the Mac Plex App is not working, even if I reinstall it:

<img width="290" alt="Bildschirmfoto 2023-09-09 um 09 40 15" src="https://github.com/lostb1t/replex/assets/144341972/98b0ec36-3464-4052-adfc-6206e3758d0b">

**Reply by lostb1t** [reply]
Well the mac app is just a browser so the same cors issue is at play. Try for example with the htpc version for the mac 

https://downloads.plex.tv/htpc/1.47.1.3971-8b24dbb6/macos/PlexHTPC-1.47.1.3971-8b24dbb6-universal.zip?_gl=1*m9hlaf*_ga*NzYxNDYzMDcuMTY4ODk4NjQ5NQ..*_ga_G6FQWNSENB*MTY5NDI0NTc5OC4xOC4xLjE2OTQyNDU4MTAuNDguMC4w

Either way need to figure out howto fix cors in your stack. Seems nginx is adding a cors value and plex aswell. Need to disable it from nginx

**Reply by lostb1t** [reply]
Try setting proxy_pass_request_headers on; in your server block

I'll see if I can setup nginx proxy manager this weekend

**Comment by cine25** [comment]
I will try it with this setting in nginx config:

add_header Access-Control-Allow-Origin "example1.com";
add_header Access-Control-Allow-Origin "example2.com";

maybe it will work. if its not working I will try another plex instance on another server without pre configured domain (complete root access). I will tell if it's working :)

**Comment by Zorro92** [comment]
If you don't know how to use nginx, don't use it. Use caddy it's much simpler and it automatically handles certificates for you for https.
